compiler/src/codegen/compcore.re

```diff
@@ -3984,6 +3984,13 @@ let compile_wasm_module = (~env=?, ~name=?, prog) => {
   let () = ignore @@ compile_tables(wasm_mod, env, prog);
   let () = ignore @@ compile_elems(wasm_mod, env, prog);
 
+  let serialized_cmi = Cmi_format.serialize_cmi(prog.signature);
+  Module.add_custom_section(
+    wasm_mod,
+    "cmi",
+    Bytes.to_string(serialized_cmi),
```

Comment:
Any particular reason this API takes a string instead of bytes? The JS API takes a uint8array.

Reply:
Just because when I wrote it, I figured that it would always be a string. The [C API called it `char *`](https://github.com/WebAssembly/binaryen/blob/master/src/binaryen-c.h#L2322) so my brain just went straight to strings. I would be down to change that though.